jax/experimental/jax2tf/tests/call_tf_test.py

```diff
@@ -737,6 +766,56 @@ def fun_tf(x):
     ):
       _ = fun_jax_3(x)
 
+  def test_multi_platform(self):
+    def tf_fun(x):
+      return tf.math.sin(x)
+
+    def f_jax(x):
+      return jnp.cos(jax2tf.call_tf(tf_fun)(jnp.cos(x)))
+    x = np.arange(12, dtype=np.float32).reshape((3, 4))
+
+    # TODO(b/306753579): call_tf can only be lowered when we have a device
+    lowering_platforms = tuple(
+      p if p != "gpu" else "cuda"
+      for p in self.__class__.jax_and_tf_platforms)
+
+    exp = export.export(f_jax,
+                        lowering_platforms=lowering_platforms)(x)
+    for jax_platform in self.__class__.jax_and_tf_platforms:
+      jax_device = jax.devices(jax_platform)[0]
+      x_device = jax.device_put(x, jax_device)
+      logging.info("Running harness natively on %s", jax_device)
+      native_res = f_jax(x_device)
+      logging.info("Running exported harness on %s", jax_device)
+      exported_res = export.call_exported(exp)(x_device)
+      self.assertAllClose(native_res, exported_res)
+
+  def test_multi_platform_call_tf_graph(self):
+    def tf_fun(x):
+      return tf.math.sin(x)
+
+    def f_jax(x):
+      return jnp.cos(jax2tf.call_tf(tf_fun,
+                                    call_tf_graph=True,
+                                    ordered=True)(jnp.cos(x)))
+    x = np.arange(12, dtype=np.float32).reshape((3, 4))
+    # When we use call_tf_graph we can serialize for multiple platforms
+    lowering_platforms = ("tpu", "cpu", "cuda")
+    # We must use jax2tf.convert to run a call_tf(call_tf_graph)
+    # TODO(necula): if we remove the tf.function and we have multiple platforms
+    # then we attempt to lower call_tf multiple times and only the first
+    # lowering will have the proper side effects for the function_list.
+    f_tf = tf.function(jax2tf.convert(
+      f_jax,
+      native_serialization=True,
+      native_serialization_platforms=lowering_platforms))
+    for tf_device in self.__class__.tf_devices:
+      logging.info(
```

Comment:
Consider making this a subtest: https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests.